libraries/botframework-connector/src/auth/jwtTokenExtractor.ts

```diff
@@ -94,7 +96,7 @@ export class JwtTokenExtractor {
         const keyId: string = decodedToken.header.kid;
         const metadata: any = await this.openIdMetadata.getKey(keyId);
         if (!metadata) {
-            throw new Error('Signing Key could not be retrieved.');
+            throw new AuthenticationError('Signing Key could not be retrieved.', StatusCodes.NOT_FOUND);
```

Comment:
Does this end in an unauthorized request? If yes, UNAUTHORIZED could be a more appropriate status code.

Reply:
i.e. if the request ends here, we want to return an http unauthorized and not a 400.

Reply:
Hmm...
I suppose it would _ultimately_ end up being unauthorized, because even though we can `decode` the token initially and get a decoded token, `decode` doesn't verify if the JWT signature is valid. (decode to find out what key we need, then we call `verify` to actually get a decoded token while also verifying if signature is valid)

So I guess trying to just `getKey(keyId)` might be not found (we're just asking if the metadata has a key with the specific id), but the ultimate result in the encompassing `validateToken` method semantically should be that the client is unauthorized.

Changing to UNAUTHORIZED 👍